### ISSUE TITLE
Add test for pesmd

### ISSUE THREAD

**Issue by GiovanniBussi** [plumed/plumed2#482]
@gtribello could you add a test for pesmd? It looks [there is none](https://www.plumed.org/coverage-v2.5/coverage/cltools/pesmd.cpp.gcov.html).

Also please check the merge I did on v2.4/v2.5/master right now. It looked like you merged separate changes on v2.4/master and on v2.5.

Thanks

Giovanni

**Comment by gtribello** [comment]
Hello @GiovanniBussi 

Thanks for this message.  I did exactly what you said in terms of doing the separate changes (sorry).  What should I do to do this properly?  Should I make the change on v2.4 merge v2.4 into v2.5 and then merge v2.5 into master?  Alternatively, should I just make the change on one of the three branches and then cherry-pick it into the others.

With regards to adding a test, I am having a bit of difficulty and I wondered if you could help.  I want to run the command:

plumed pesmd < input

I thus created a config that looks like this:

type=plumed
arg="pesmd < input"

when I try to run make on this regtest though it hangs.  It looks to me like the direct in is not working i.e. it is waiting to read in the input from standard input and ignoring the file that I have told it to read.  I figure this probably happens because of some special-character bullshit in bash or make that I don't know.  Do you have any idea as to what I am doing wrong?

**Comment by valsson** [comment]
Does it work using just the following? 

type=plumed
arg="pesmd input"

I have something similar in regtest/ves/rt-md_le-2d and using this is in config file works

type=plumed
arg="ves_md_linearexpansion input"

Omar

**Comment by GiovanniBussi** [comment]
@gtribello I think the standard way would be to make the fix in 2.4 then merge to 2.5 and than to master. In this way if later someone else merges v2.4 in 2.5 git knows that change is already present. If you cherry pick the change, git should be able to see that an identical change was applied and will ignore it.

One problem of your fix I think is that you used `int` variables in some case and `unsigned` in another, so they are considered as different.

I merged v2.4 to v2.5 and v2.5 to master using `git merge -s ours` that basically ignores the commit in the upstream branch so that later it is known to be already present. Anyway, if now I do
````
git diff v2.5 master pesmd.cpp
````
I see a difference
````
-        if( 3*i+j>dim-1 ) break;
+        if( 3*i+j>dim ) break;
````

Could you double check that the three branches (v2.4/v2.5/master) are correct now?

About the test I think what @valsson suggested should work.

I think that adding a test in v2.4 then merging it to v2.5 and then to master would be ideal so that we know that  the same test works equally in the three branches (if the features did not change).

Thanks!

Giovanni

**Comment by gtribello** [comment]
I fixed this.  Probably could have been cleaner with the commits but hey lesson learnt for next time.

**Comment by GiovanniBussi** [comment]
See 2d099818165b093e9e1aa95bf4e4846095005849

**Comment by gtribello** [comment]
Hello @GiovanniBussi

I have tried to fix this in a branch that I have built off v2.4.  The new branch is called am-I-loosing-my-mind.  It is very much a question and if it passes on travis then the answer is yes as I genuinely don't have any clue what I did in "fixing" this the other day.

If I merge your change to v2.4 into my local branch will I lose the regtest as you reverted the previous commit?  In other words, what should I do here.  I am thinking:

(1)  Save all the files for the regtest somewhere else
(2)  git pull origin v2.4
(3)  git add <all the files in the regtest>
(4)  git commit 

Will that work?
